images/virt-artifact/patches/024-cover-kubevirt-metrics.patch

```diff
@@ -0,0 +1,165 @@
+ 	flag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+ 	flag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("kubeconfig"))
+@@ -61,6 +67,9 @@ func (service *ServiceListen) InitFlags() {
+ func (service *ServiceListen) AddCommonFlags() {
+ 	flag.StringVar(&service.BindAddress, "listen", service.BindAddress, "Address where to listen on")
+ 	flag.IntVar(&service.Port, "port", service.Port, "Port to listen on")
++	// default values are taken from the common server counterparts
++	flag.StringVar(&service.MetricsBindAddress, "metrics-listen", service.BindAddress, "Address for metrics to listen on")
++	flag.IntVar(&service.MetricsPort, "metrics-port", service.Port, "Port for metrics to listen on")
+ }
+ 
+ func (service *ServiceLibvirt) AddLibvirtFlags() {
+diff --git a/pkg/virt-operator/resource/generate/components/daemonsets.go b/pkg/virt-operator/resource/generate/components/daemonsets.go
+index da6e00c783..f2484ebe2d 100644
+--- a/pkg/virt-operator/resource/generate/components/daemonsets.go
++++ b/pkg/virt-operator/resource/generate/components/daemonsets.go
+@@ -175,6 +175,10 @@ func NewHandlerDaemonSet(namespace, repository, imagePrefix, version, launcherVe
+ 	container.Args = []string{
+ 		"--port",
+ 		"8443",
++		"--metrics-listen",
++		"127.0.0.1",
++		"--metrics-port",
++		"8080",
+ 		"--hostname-override",
+ 		"$(NODE_NAME)",
+ 		"--pod-ip-address",
+@@ -192,7 +196,7 @@ func NewHandlerDaemonSet(namespace, repository, imagePrefix, version, launcherVe
+ 		{
+ 			Name:          "metrics",
+ 			Protocol:      corev1.ProtocolTCP,
+-			ContainerPort: 8443,
++			ContainerPort: 8080,
+ 		},
+ 	}
+ 	container.SecurityContext = &corev1.SecurityContext{
+@@ -226,7 +230,7 @@ func NewHandlerDaemonSet(namespace, repository, imagePrefix, version, launcherVe
+ 		FailureThreshold: 3,
+ 		ProbeHandler: corev1.ProbeHandler{
+ 			HTTPGet: &corev1.HTTPGetAction{
+-				Scheme: corev1.URISchemeHTTPS,
++				Scheme: corev1.URISchemeHTTP,
+ 				Port: intstr.IntOrString{
+ 					Type:   intstr.Int,
+ 					IntVal: 8443,
+@@ -241,7 +245,7 @@ func NewHandlerDaemonSet(namespace, repository, imagePrefix, version, launcherVe
+ 	container.ReadinessProbe = &corev1.Probe{
+ 		ProbeHandler: corev1.ProbeHandler{
+ 			HTTPGet: &corev1.HTTPGetAction{
+-				Scheme: corev1.URISchemeHTTPS,
++				Scheme: corev1.URISchemeHTTP,
+ 				Port: intstr.IntOrString{
+ 					Type:   intstr.Int,
+ 					IntVal: 8443,
+diff --git a/pkg/virt-operator/resource/generate/components/deployments.go b/pkg/virt-operator/resource/generate/components/deployments.go
+index c86493e95f..c05d274db9 100644
+--- a/pkg/virt-operator/resource/generate/components/deployments.go
++++ b/pkg/virt-operator/resource/generate/components/deployments.go
+@@ -75,7 +75,7 @@ func NewPrometheusService(namespace string) *corev1.Service {
+ 					Port: 443,
+ 					TargetPort: intstr.IntOrString{
+ 						Type:   intstr.String,
+-						StrVal: "metrics",
++						StrVal: "https-metrics",
```

Comment:
Why https? You are remove tls

Reply:
This is a config for Service `kubevirt-prometheus-metrics`, not for container port inside the `virt-handler` pod. We need Prometheus to collect metrics from the kube-rbac-proxy port, which is labeled as `https-metrics`.